### DISCUSSION TITLE
[SC]: Consultazione ANSC Intestatario - Data nascita del ModelSoggetto non formattata correttamente.

### DISCUSSION THREAD

**Discussion by gcuci** [italia/ansc#832] [Miglioramenti] [answered]
### Ambiente

pre-produzione

### Servizio

R005 - Servizi cooperativi consultazione

### idOperazione

2168108

### Provenienza segnalazione

Software House

### Comune di riferimento e operatore

Bagnacavallo

### Data del test (yyyy-mm-dd)

2024-04-12

### Descrivi il problema

Il servizio di Consultazione ANSC Intestatario, restituisce un ModelSoggetto non coerente alla documentazione (swagger) pubblicata.

Ci sono delle proprietà non previste, 'datains' e 'dataultimoagg' (ma questo potrebbe non essere un problema), e la proprietà dataNascita, non è nel formato previsto. Viene restituita una data nel formato data/ora ISO, e non nel formato previsto dalla documentazione, 'AAAA-MM-DD'. Questo ci causa un'eccezione di parsing del formato della data, con le classi del Client, generate a partire dallo swagger.

Per un esempio in pre-produzione, potete far riferimento all'idOperazione: '2168108', che ha restituito questa Response (dati fittizi):

{
    "testataRisposta": {
        "idComune": 595,
        "idOperazioneComune": "11392",
        "idOperazione": "2168108",
        "idEsito": 0
    },
    "datiPaginazione": {
        "numPagina": 1,
        "numRecordPagina": 10,
        "numRecordTotali": 0
    },
    "infos": [
        {
            "code": "110527",
            "severity": "I",
            "text": "Informazioni sull'operazione da fornire in caso di assistenza : data:2024-04-12 13:30:30 , idOperazione:2168108"
        }
    ],
    "soggetti": [
        {
            "id": 2301868,
            "cognome": "DONNA2",
            "nome": "UC",
            "sesso": "F",
            "datains": "2024-03-18T11:23:30.000+00:00",
            "dataultimoagg": "2024-03-18T12:27:23.000+00:00",
            "codiceFiscale": "DNNCUX90A41A547L",
            "idComuneNascita": 595,
            "idStatoNascita": "90",
            "idComuneResidenza": 9562,
            "nazionalita": "ITALIANA",
            "nomeStatoNascita": "ITALIA",
            "idProvinciaNascita": 39,
            "idProvinciaResidenza": 39,
            "idNazionalita": 90,
            "dataNascita": "1990-01-01T00:00:00.000+00:00",
            "idStatoResidenza": "90"
        }
    ]
}

da cui si evince il formato della 'dataNascita'.

**Comment by gbuemi** [comment]
Buongiorno @gcuci,
trasformiamo la issue in discussion e seguiranno aggiornamenti in merito.

Un saluto,
Supporto ANSC

**Comment by mttfranci** [comment, marked as answer]
@gcuci  con il rilascio della patch [1.27.6 ](https://github.com/italia/ansc/releases/tag/v1.27.6)abbiamo introdotto un meccanismo di retrocompatibilità con cui potete attivare la risposta corretta.

Dovrebbe essere sufficiente passare al path segnalato il parametro [revision](https://github.com/italia/ansc/blob/main/docs/Revision.md)=2000002, ad esempio : 

`/consultazione/ansc/intestatario/1?revision=2000002`

In tal caso l'output restituito dovrebbe essere del tipo : 

```
        {
            "id": "567260",
            "codiceFiscale": "XXXYYY90F50H501G",
            "cognome": "UTENTE",
            "nome": "TEST",
            "sesso": "F",
            "dataNascita": "1991-10-10",
            "idProvinciaNascita": "63",
            "idStatoNascita": "90",
            "nomeStatoNascita": "ITALIA",
            "idComuneResidenza": "2780",
            "idStatoResidenza": "90",
            "idNazionalita": "90",
            "documentoRiconoscimento": {}
        }
```

Ovvero aderente alle specifiche openapi del servizio [R005](https://github.com/italia/ansc/blob/main/docs/openapi/R005_consultazione_ansc.yaml)

Trovate qualche dettaglio in più nel documento <https://github.com/italia/ansc/blob/main/docs/Revision.md>
Siamo a disposizione per ogni ulteriore chiarimento.

Saluti

Assistenza ANSC
